src/main/java/docbuddy/users/persistence/dao/UserDaoImpl.java

```diff
@@ -134,4 +154,14 @@ public User login(User user) {
         }
         return resultUsers;
     }
+
+    private void validateUser(User user) {
+        if (user.getUserName() == null || user.getUserName().isEmpty()) {
```

Comment:
Maybe we can use Google Guava or Apache Commons String validation, they are already null zafe:

https://stackoverflow.com/questions/14721397/checking-if-a-string-is-empty-or-null-in-java

Reply:
Absolutely right. Done.